### DISCUSSION TITLE
Is there possible to use config.leader with key repeatly?

### DISCUSSION THREAD

**Discussion by zw963** [wezterm/wezterm#4927] [Q&A] [answered]
### What Operating System(s) are you running on?

Linux Wayland

### WezTerm version

20240128-202157-1e552d76

### Ask your question!

I use `config.leader` like this: it cool!

```
config.leader = { key = 'x', mods = 'CTRL' }
config.keys = {
  {
    key = 'RightArrow',
    mods = 'LEADER',
    action = act.AdjustPaneSize { 'Right', 5 }
  },
  {
    key = 'LeftArrow',
    mods = 'LEADER',
    action = act.AdjustPaneSize { 'Left', 5 }
  },
  {
    key = 'UpArrow',
    mods = 'LEADER',
    action = act.AdjustPaneSize { 'Up', 5 }
  },
  {
    key = 'DownArrow',
    mods = 'LEADER',
    action = act.AdjustPaneSize { 'Down', 5 }
  },
```

But, there is a issue, adjust pane size action only one shot, e.g. `C-x ->`, will expand current pane right 5px,  When i try `C-x -> ->`, 
only the first time -> works, the latter -> just ignore, what i expected is, `C-x -> -> ` will expand current pane right twice, 10px, is there possible?

I search in document for a while, don't found a option for this.

Thanks.

**Comment by bew** [comment]
Yes, you need to setup a Key Table for your leader keymap (the existing leader key system predates the key tables, which are much more flexible)

And use a ActivateKeyTable action with specific activation options to get the behavior you want.

You can find a full example in https://wezfurlong.org/wezterm/config/key-tables.html

**Reply by zw963** [reply]
Thanks for you answer!

i saw the `one_shot = false` option before, it's looks like this option really used for this purpose.

But, it still not solve my issue,  what i expected is keybinding like `C-x -> -> ->`, leader (Ctrl + x) with table directly, without key.

Check following code:

```lua
config.leader = { key = 'x', mods = 'CTRL' }

config.keys = {
  {
    key = '',  -- How to set this to a empty key?
    mods = 'LEADER',
    action = act.ActivateKeyTable {
      name = 'resize_pane',
      one_shot = false
    },
  },
 }

config.key_tables = {
  resize_pane = {
    { key = 'LeftArrow', action = act.AdjustPaneSize { 'Left', 5 } },

    { key = 'RightArrow', action = act.AdjustPaneSize { 'Right', 5 } },

    { key = 'UpArrow', action = act.AdjustPaneSize { 'Up', 5 } },

    { key = 'DownArrow', action = act.AdjustPaneSize { 'Down', 5 } },

    -- Cancel the mode by pressing escape
    { key = 'Escape', action = 'PopKeyTable' },
  },
}
```

Thanks

**Reply by bew** [reply, marked as answer]
You can forget about `config.leader`, and do something like:
```lua
config.keys = {
  {
    mods = 'CTRL',
    key = 'x',
    action = act.ActivateKeyTable {
      name = 'resize_pane',
      one_shot = false
    },
  },
 }

 -- ...
```

**Comment by zw963** [comment]
Okay, i get my answer:

```lua
config.leader = { key = 'x', mods = 'CTRL' }
{ key = 'x', mods = 'LEADER|CTRL', action = act.SendKey { key = 'x', mods = 'CTRL' } },
```

When i pressing `C-x C-x`, it will send `C-x`, this is what i want. thanks

**Comment by zw963** [comment]
Yeah!  i solved my issue completely!

following is config:

```
  {
    key = 'e',
    mods = 'LEADER|CTRL',
    action = act.Multiple {
      act.SendKey { key = 'x', mods = 'CTRL' },
      act.SendKey { key = 'e', mods = 'CTRL' }
    }
  },
```

Thank you very much!
